app/controllers/api/migration_analytics_manifest_controller.rb

```diff
@@ -0,0 +1,34 @@
+def parse_json_if_valid(json)
+    return JSON.parse(json)
+  rescue JSON::ParserError => e
+    return :invalid
+end
+
+module Api
+  class MigrationAnalyticsManifestController < BaseController
+    def index
+      tmp_manifest_path = '/tmp/migration-analytics-manifest.json'
+      tmp_manifest = File.file?(tmp_manifest_path) ? parse_json_if_valid(File.read(tmp_manifest_path)) : nil
+      tmp_manifest_valid = tmp_manifest != nil && tmp_manifest != :invalid
+
+      if tmp_manifest_valid
+        manifest_path = tmp_manifest_path
+        manifest = tmp_manifest
+      else
+        manifest_path = File.join(
+          Bundler.environment.specs['cfme-migration_analytics'].first.full_gem_path,
+          'config/default-manifest.json'
+        )
+        manifest = parse_json_if_valid(File.read(manifest_path))
+      end
+
+      res = {
+        :tmp_manifest_present => tmp_manifest != nil,
+        :tmp_manifest_valid => tmp_manifest_valid,
+        :path => manifest_path,
+        :body => manifest,
+      }
+      render_resource :migration_analytics_manifest, res
+    end
+  end
+end
```

Comment:
we need specs added for the API. so spec/requests/migration_analytics_manifest_spec.rb, maybe look at https://github.com/RedHatCloudForms/cfme-cloud_services/pull/23/files for spec/ files for the api file plus the spec_helper.rb and support/api_request_helpers.rb.

Reply:
Ok, I'll take a look and try to get these started. I got stuck trying to get the Travis config's Ruby checks to work in this repo though, I may need some help figuring that out (the bin/setup script fails, probably because it was depending on something from https://github.com/martinpovolny/miq_plugin_example that I removed). I have an issue to track that: #2